DCOS-21520: check node version on checkout

juliangieseke opened the pull request:
This adds a small notice after you changed the branch, this should come in handy once we introduced the node upgrade but still have to work on older releases 👍 

Example with correct Node version:
![screen shot 2018-03-02 at 17 15 22](https://user-images.githubusercontent.com/530632/36909182-9f5457ba-1e3d-11e8-8817-1e46be3da786.png)
And wrong one:
![screen shot 2018-03-02 at 17 20 18](https://user-images.githubusercontent.com/530632/36909338-038c813a-1e3e-11e8-8fe6-9c747e3ce0cc.png)

DanielMSchmidt commented on scripts/pre-install:
```diff
@@ -12,6 +12,6 @@ source ${SCRIPT_PATH}/utils/marathon
 ${SCRIPT_PATH}/validate-engine-versions && \
   install_marathon_raml "$MARATHON_VERSION" && \
-  install_hooks "pre-commit post-rewrite post-commit"
+  install_hooks "pre-commit post-rewrite post-commit post-checkout"
```
Nice to know there is something like post-checkout :)

DanielMSchmidt commented on scripts/post-checkout:
```diff
@@ -0,0 +1,22 @@
+    fi
+fi
+
+info "Looks like your node version is fine."
```
I think bash scripts have the convention to end with a new line

DanielMSchmidt commented on scripts/post-checkout:
```diff
@@ -0,0 +1,22 @@
+NODE_VERSION=$(node -v)
+
+if [ "$(node -p -e "require('./package.json').devDependencies.node")" != "undefined" ]; then
+    if [ "$(echo $NODE_VERSION | | cut -d. -f1)" -lt 6 ]; then
```
As far as I understand after looking at it for 10 minutes this checks for the major version. Could you move this into a variable? `$MAJOR_NODE_VERSION -lt 6` would be nicer to read IMHO

juliangieseke commented:
updated the code to contain more variables, it should be way easier to understand now 👍 

also I changed the check a bit, it now checks vor `npm -v` instead of `node -v` - reason is, that `npm` has to be up to date in order to support `node` as a dependency (also it has to be a decent node version, but that should be assured in the package.json, not here…
